spec/services/subscriber_list_creator_spec.rb

```diff
@@ -0,0 +1,95 @@
+require "rails_helper"
+
+RSpec.describe SubscriberListCreator do
+  context "when gov delivery ID is not passed in" do
+    let(:params) do
+      {
+        title: "This is a short title",
+        enabled: true,
+        tags: {},
+        links: {},
+        migrated_from_gov_uk_delivery: false,
+        document_type: 'news-article',
+        email_document_supertype: 'publications',
+        government_document_supertype: 'news-articles',
+      }
+    end
+
+    before do
+      allow(Services.gov_delivery).to receive(:create_topic).and_return('TOPIC_123')
+    end
+
+    it "creates a subscriber list with title, containing the record ID" do
+      creator = described_class.new(params)
+      creator.save
+      expect(creator.record).to have_attributes(
+        title: "This is a short title (#{creator.record.id})",
+        enabled: true,
+        tags: {},
+        links: {},
+        migrated_from_gov_uk_delivery: false,
+        document_type: 'news-article',
+        email_document_supertype: 'publications',
+        government_document_supertype: 'news-articles',
+        gov_delivery_id: 'TOPIC_123',
+      )
+    end
+
+    it "creates the topic on gov_delivery" do
+      creator = described_class.new(params)
+      creator.save
+      expect(Services.gov_delivery).to have_received(:create_topic)
+        .with("This is a short title (#{creator.record.id})")
+    end
+
+    context "and the title is longer than 255 characters" do
+      it "trims the title" do
+        title = 260.times.map { ('A'..'Z').to_a.sample }.join
```

Comment:
I think this could do with being simpler. It doesn't have to be a randomly generated string. A static value (eg `'A' * 260) would do the job and make the expectation below a touch easier to read too.